sa/saro.go

```diff
@@ -994,6 +994,16 @@ func (ssa *SQLStorageAuthorityRO) GetValidOrderAuthorizations2(ctx context.Conte
 		return nil, errIncompleteRequest
 	}
 
+	// The authz2 and orderToAuthz2 tables both have a column named "id", so we
+	// need to be explicit about which table's "id" column we want to select.
+	qualifiedAuthzFields := strings.Split(authzFields, " ")
```

Comment:
This seems really fragile since it depends on the authzFields being comma _and_ space separated "a, b, c". Since we'll need to do a join here anyhow, it would be better to store authzFields as a slice and join both here and elsewhere that it's used.

Alternately - can we unconditionally qualify id? That is, make it always say `authz2.id`?

Reply:
We don't currently store authzFields as a slice because slices can only be `var`s, while the string version is currently a `const`.

But yeah, I'm happy to just build the table name into the string itself.

Reply:
Ah hmm, actually this fails. Due to https://github.com/letsencrypt/boulder/blob/914e971f15589b33689f005ea1015b381e3d2c9b/sa/sa.go#L429 in which "authz2.id" is an "invalid MariaDB identifier". And which also relies on the comma-space delimitation of the authzFields string, incidentally.